### PR TITLE
Clean up HFX energy routines

### PR DESCRIPTION
**Pull request by vanderhe**
Clean up as discussed in the devel. meeting.

### CODE REVIEW

**Comment by vanderhe** [comment]
@bhourahine I don't see any reason why the HFX energy should be queried by `getPositionDependentEnergy()`, since `calcEnergies()` inside of `getTDEnergy()` already calculates it. Any objections?

**Comment by bhourahine** [comment]
> @bhourahine I don't see any reason why the HFX energy should be queried by `getPositionDependentEnergy()`, since `calcEnergies()` inside of `getTDEnergy()` already calculates it. Any objections?

I think I agree that it should be removed.
@fbonafe, any comments?

File: src/dftbp/timedep/timeprop.F90 [hunk 1]
```diff
@@ -2030,7 +2030,7 @@ subroutine getTDEnergy(this, env, energy, rhoPrim, rho, neighbourList, nNeighbou
     ! Calculate nuclear kinetic energy
     energyKin = 0.0_dp
     if (this%tIons) then
-      energyKin = 0.5_dp * sum(this%movedMass(:,:) * this%movedVelo(:,:)**2)
+      energyKin = 0.5_dp * sum(this%movedMass * this%movedVelo**2)
```
Comment (bhourahine) [inline src/dftbp/timedep/timeprop.F90:2033]: Slightly off topic for this PR, but harmless and needed.

File: src/dftbp/timedep/timeprop.F90 [hunk 2]
```diff
@@ -3733,7 +3733,7 @@ end subroutine initLatticeVectors
 
   !> Calculates repulsive and dispersion energies
```
Comment (bhourahine) [inline src/dftbp/timedep/timeprop.F90:3734]: This comment certainly suggests hybrid energy evaluation doesn't belong here.

**Review by bhourahine — APPROVED** [review]
LGTM

**Pull request merged**